testsuite/tests/kuadrant/gateway/reconciliation/test_affected_by.py

```diff
@@ -0,0 +1,31 @@
+"""Tests that affected by status is applied correctly to the HTTPRoute and Gateway"""
+
+import pytest
+
+pytestmark = [pytest.mark.kuadrant_only]
+
+
+def test_route_status(route, rate_limit, authorization):
+    """Tests affected by status for HTTPRoute"""
+    route.refresh()
+    assert route.is_affected_by(rate_limit)
+    assert route.is_affected_by(authorization)
+
+    rate_limit.delete()
+    assert not route.refresh().is_affected_by(rate_limit)
+
+    authorization.delete()
+    assert not route.refresh().is_affected_by(authorization)
+
+
+def test_gateway_status(gateway, dns_policy, tls_policy):
+    """Tests affected by status for Gateway"""
+    gateway.refresh()
+    assert gateway.is_affected_by(dns_policy)
+    assert gateway.is_affected_by(tls_policy)
+
+    dns_policy.delete()
+    assert not gateway.refresh().is_affected_by(dns_policy)
```

Comment:
this test may easily become flaky since the status is eventually consistent. I suggest adding backoff or better something like wait_for_condition() to have it hardened by default.